mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/DefaultQuotaRootResolver.java

```diff
@@ -55,6 +55,7 @@ public QuotaRoot createQuotaRoot(String quotaRootValue) {
     @Override
     public QuotaRoot getQuotaRoot(MailboxPath mailboxPath) throws MailboxException {
         Preconditions.checkArgument(!mailboxPath.getNamespace().contains(SEPARATOR), "Namespace should not contain " + SEPARATOR);
+        Preconditions.checkArgument(mailboxPath.getUser() != null, "Username should not be null");
```

Comment:
Hi Edgar.

Thank you very much for your contribution.

However, I don't think solving this issue is that easy... `null` user of a mailboxpath had been used for shared folders thus I think we should consider it as valid.

What would you thing of this:

```
MailboxPath("#namespace", "user", "any) -> QuotaRoot will be "#namespace&user"
MailboxPath("#namespace",  null, "any) -> QuotaRoot will be "#namespace"
```
(We just ignore the user part of the quota root when none).

Of course, we would need to better represent the fact that a mailbox path might not have a user (by using Optional?) but that is a long refactoring on a central object that would introduce API breaking changes, hence the former solution seems better to me.

Best regards,

Benoit

Reply:
@chibenwa I checkout `master` and did not found the `DefaultQuotaRootResolver`. The `3.0.1-RC` is away behind the `master`, so I suspected that this change don't event fit today's requirements.

Reply:
Don't you mind opening this pull request targetting master?

We plan to release 3.1.0 soon. 

Is it a bug you encoutered on 3.0.1 or did you just followed the JAMES-1840 ticket?

Reply:
@chibenwa Just followed the ticket.

Reply:
The ticket states that null user values should be accepted (and not rejected). Though I agree this ticket is hardly readable.... I hope I made progress at writing tickets since 2016 ^^

Sorry for the inconvenience.

Reply:
You might need to rebase and remove unrelated comments.

Reply:
Indeed it should accept the `null`s. Will update the PR soon.